Python/Product/Analysis/ModuleAnalysis.cs

```diff
@@ -710,6 +692,40 @@ private class ErrorWalker : PythonWalker {
             return res;
         }
 
+        internal IEnumerable<MemberResult> GetAllAvailableMembersFromScope(InterpreterScope scope, GetMemberOptions options) {
+            var result = new Dictionary<string, IEnumerable<AnalysisValue>>();
+            var scopeResult = GetAllAvailableAnalysisValuesFromScope(scope, options);
+            foreach (var kvp in scopeResult) {
+                result[kvp.Key] = kvp.Value;
+            }
+
+            var res = MemberDictToResultList(GetPrivatePrefix(scope), options, result);
+            if (options.StatementKeywords() || options.ExpressionKeywords()) {
+                res = GetKeywordMembers(options, scope).Union(res);
+            }
+            return res;
+        }
+
+        private Dictionary<string, List<AnalysisValue>> GetAllAvailableAnalysisValuesFromScope(InterpreterScope scope, GetMemberOptions options) {
+            var scopeResult = new Dictionary<string, List<AnalysisValue>>();
+            foreach (var kvp in scope.GetAllMergedVariables()) {
+                var vars = kvp.Value.TypesNoCopy;
+                if (options.Exceptions() && !IsExceptionType(kvp.Key, vars)) {
+                    continue;
+                }
+                if (scopeResult.TryGetValue(kvp.Key, out var values)) {
+                    values.AddRange(vars);
+                } else {
+                    scopeResult[kvp.Key] = values = new List<AnalysisValue>(vars);
+                }
+                values.Add(new SyntheticDefinitionInfo(
+                    kvp.Key,
+                    null,
+                    kvp.Value.Definitions.Select(e => e.GetLocationInfo())
+                ));
```

Comment:
Do we need this if there have been variables added? Or do we need it if the value has no definitions?

Reply:
It was existing code that applied to a single scope and now called for each scope recursively.